tbot/main.py

```diff
@@ -57,11 +59,32 @@ def _import_hightlighter() -> typing.Callable[[str], str]:
 # }}}
 
 
+class TbotArgumentParser(argparse.ArgumentParser):
+    def convert_arg_line_to_args(self, arg_line: str) -> List[str]:
+        """
+        Make it possible to use shell variables also in argumentsfiles.
+        """
+        if "$" in arg_line:
+            vl = re.findall(r"(\$\{.*?\})", arg_line)
+            for v in vl:
+                varname = v[2:-1]
+                env = os.getenv(varname)
+
+                if env is None:
+                    raise RuntimeError(f"Could not find Environment variable {varname}")
```

Comment:
I'm wondering whether it is better to raise an exception or to expand the variable to nothing, similar to how shells are doing it...

On one hand, in most situations, a missing variable means someone messed up and an explicit exception will notify them of it much faster.

On the other side, there may be use-cases where people expect the variables in the argument file to behave exactly like they behave on the command-line...

Reply:
I use this for example in gitlab CI environment, and a missing variable is
mostly an error, because I want to setup something .... and if I forgot it, it
would be nice to see it immediately... and stop, as it makes no sense to go
further... okay, may we just print a log message, but it may takes a lot of time
to findout testcases X fails because variable is empty...

Reply:
Fair, let's go with the exception version then :+1: